src/streamlink/plugins/nrk.py

```diff
@@ -35,30 +15,73 @@
     )
 })
 
+_playable_schema = validate.Schema({
+    "playable": validate.all(
+        {
+            "assets": validate.all(
+                [{
+                    "url": validate.url(
+                        scheme="http",
+                        path=validate.any(
+                            validate.endswith(".m3u8"),
+                            validate.endswith(".mp3")
+                        ),
+                    ),
+                    "format": validate.all(validate.text),
+                }]
+            ),
+        }
+    ),
+    "statistics": {
+        "luna": {
+            "data": {
+                "title": validate.text,
+            },
+        },
+    },
+})
+
 
 class NRK(Plugin):
+    _psapi_url = 'https://psapi.nrk.no'
+
     @classmethod
     def can_handle_url(self, url):
         return _url_re.match(url)
 
     def _get_streams(self):
-        # Get the stream type from the url (tv/radio).
-        stream_type = _url_re.match(self.url).group(1).upper()
-        cookie = {
-            "NRK_PLAYER_SETTINGS_{0}".format(stream_type): COOKIE_PARAMS
-        }
-
-        # Construct API URL for this program.
-        baseurl = self.session.http.get(self.url, cookies=cookie, schema=_schema)
-        program_id = _id_re.search(self.url).group(1)
+        # Construct manifest URL for this program.
+        program_type, program_id = _id_re.search(self.url).groups()
+        if program_type == 'direkte':
+            manifest_type = 'channel'
+        elif program_type == 'serie':
+            manifest_type = 'program'
+        elif program_type == 'podkast':
+            manifest_type = 'podcast'
```

Comment:
It's probably better to use a key-value map here for readability reasons. Maybe also return `None` early if there's no match, in case the URL regex gets changed and the programm type checks do not get updated.

Reply:
I put this in a class attribute dict, and also changed it to raise PluginError with a descriptive message if `program_type` is unknown, which I assume is better than silently failing.